src/cursor/cur_version.c

```diff
@@ -0,0 +1,362 @@
+
+    ret = WT_NOTFOUND;
+
+    /* The cursor should be positioned. */
+    if (!F_ISSET(cursor, WT_CURSTD_KEY_INT))
+        goto err;
+
+    upd = version_cursor->next_upd;
+
+update_chain:
+    if (upd != NULL && !F_ISSET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED)) {
+        /*
+         * If the update is an aborted update, we want to skip to the next update immediately or get
+         * the ondisk value if the update is the last one in the update chain.
+         */
+        if (upd->txnid == WT_TXN_ABORTED) {
+            upd = upd->next;
+            if (upd != NULL)
+                goto update_chain;
+            else {
+                F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+                goto disk_image;
+            }
+        }
+        /*
+         * If the update is a tombstone, we still want to record the stop information but we also
+         * need traverse to the next update to get the full value. If the tombstone was the last
+         * update in the update list, retrieve the ondisk value.
+         */
+        version_cursor->upd_txnid = upd->txnid;
+        version_cursor->upd_durable_stop_ts = upd->durable_ts;
+        version_cursor->upd_stop_ts = upd->start_ts;
+        if (upd->type == WT_UPDATE_TOMBSTONE) {
+            upd = upd->next;
+            if (upd == NULL) {
+                F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+                goto disk_image;
+            }
+        }
+
+        /*
+         * Set the version cursor's key, which contains all the record metadata for that particular
+         * version of the update.
+         */
+        __wt_cursor_set_key(cursor, upd->txnid, upd->durable_ts, upd->start_ts,
+          version_cursor->upd_txnid, version_cursor->upd_durable_stop_ts,
+          version_cursor->upd_stop_ts, upd->type, upd->prepare_state, upd->flags,
+          WT_VERSION_UPDATE_CHAIN);
+
+        /*
+         * Set the version cursor's value to be the update value. If the update is a modify,
+         * reconstruct the value.
+         */
+        if (upd->type != WT_UPDATE_MODIFY) {
+            __wt_upd_value_assign(cbt->upd_value, upd);
+            ret = 0;
+        }
+        else
+            WT_RET(__wt_modify_reconstruct_from_upd_list(session, cbt, upd, cbt->upd_value));
+
+        __wt_cursor_set_value(cursor, cbt->upd_value->buf);
+
+        upd = upd->next;
+        if (upd == NULL)
+            F_SET(version_cursor, WT_VERSION_CUR_UPDATE_EXHAUSTED);
+        goto done;
+    }
+
+disk_image:
+    /* Get the ondisk value. */
+    if (!F_ISSET(version_cursor, WT_VERSION_CUR_ON_DISK_EXHAUSTED)) {
+        WT_RET(__wt_value_return_buf(cbt, cbt->ref, &cbt->upd_value->buf, &tw));
+
+        __wt_cursor_set_key(cursor, tw.start_txn, tw.durable_start_ts, tw.start_ts,
+          tw.stop_txn != WT_TXN_MAX ? tw.stop_txn : version_cursor->upd_txnid,
+          tw.durable_stop_ts != WT_TS_MAX ? tw.durable_stop_ts :
```

Comment:
This is not correct. Use WT_TIME_WINDOW_HAS_STOP

Reply:
Use a local variable for these values and set the local variables before calling this function.